Subject: [PATCH] Fix gear not showing when rotation or RTL changed

When rotation or RTL changed, gear button's translationX should be
updated to new position.

Test: (Rotation) Swipe notification to left > Gear is showing on the
right > Rotate the device > Gear button should be visible
Test: (RTL) Swipe notification to left > Set RTL language
 > Swipe notification to right in RTL configuration
 > Gear button should be visible

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/NotificationSettingsIconRow.java b/packages/SystemUI/src/com/android/systemui/statusbar/NotificationSettingsIconRow.java
index 4a1c451..e684461 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/NotificationSettingsIconRow.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/NotificationSettingsIconRow.java
@@ -239,13 +239,27 @@
     }
 
     @Override
+    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+        super.onLayout(changed, left, top, right, bottom);
+        if (mIconPlaced) {
+            setIconLocation(mOnLeft, true /* force */);
+        }
+    }
+
+    @Override
     public void onRtlPropertiesChanged(int layoutDirection) {
-        setIconLocation(mOnLeft);
+        if (mIconPlaced) {
+            setIconLocation(mOnLeft, true /* force */);
+        }
     }
 
     public void setIconLocation(boolean onLeft) {
-        if ((mIconPlaced && onLeft == mOnLeft) || mSnapping || mParent == null
-                || mGearIcon.getWidth() == 0) {
+        setIconLocation(onLeft, false /* force */);
+    }
+
+    private void setIconLocation(boolean onLeft, boolean force) {
+        if (mParent == null || mGearIcon.getWidth() == 0
+                || (!force && ((mIconPlaced && onLeft == mOnLeft) || mSnapping))) {
             // Do nothing
             return;
         }
